lib/APPTurtle/src/MicroRosClient.cpp

```diff
@@ -198,6 +198,9 @@ bool MicroRosClient::configureClient()
     rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
     m_allocator                     = rcl_get_default_allocator();
 
+    LOG_INFO("Connecting to micro-ROS agent %s:%u ...", m_agentConfiguration.address.toString().c_str(),
```

Comment:
The pattern `toString().c_str()` ist risky.  The "toString()" method returns a temporary string that nobody uses after calling c_str(). It might get destructed right after the c_str() call, making the pointer dangling.  

Recommend to use safe pattern:

         String ipAddrStr(m_agentConfiguration.address.toString());
         LOG_INFO("Connecting to micro-ROS agent %s:%u ...",
                  ipAddrStr.c_str(),
                  m_agentConfiguration.port);